### ISSUE TITLE
Question about the supported context length

### ISSUE THREAD

**Issue by shaoyijia** [kaistAI/prometheus#6]
Hi,

This is an exciting work and thanks for open-sourcing the model!

What's the supported context length of Prometheus? I failed to find the information in your documentation. In fact, I'm trying to use Prometheus on grounded-document writing task so the output and example can be very long (with a maximum length of 10k words) so I'm unsure whether Prometheus can support this use case.

Thank you so much!

**Comment by shaoyijia** [comment]
I haven't gone so far (trying samples with 10k words) yet. But when my prompt has around 3.5k token, I get the output like this:

>  [/INST] with coherentations and scoring. The scoring scheme is aScore 1: aScore 1: The content of a coherentations and coherent; the following articles are organized; aScore 1: aScore 1: coherent;aScore 1: coherentations. The article isScore 1:score 1: organized; theScore 2: coherently scored; theScore 1:organized; aScore 1:theScore 1: organized; theScore 1: theScore the scored; the score the score and theScore 1: the scoring process.
> ation score; the score the scoring; theScore: theScore; theScore 1:the scoring 1: theScore; the score the score; the score; the score; the score; the score; the score; the score; the; the score the score; the score; the scoring.</s>

**Comment by SeungoneKim** [comment]
@shaoyijia Hello Yijia, thank you for your interest in our work!

Prometheus is based on Llama-2-Chat, so the max context length is 4096. During training, the maximum input length was 3072 and the output length was between 256 and 1024.

I think there will be issues if you directly use Prometheus for your case. You might need to check out techniques that increase the input length like AliBi [paper](https://arxiv.org/abs/2108.12409).

I'm interested in this as well, let's keep in touch!

**Comment by SeungoneKim** [comment]
For the hyperparameters, you could refer to Table 9 in the paper (Appendix C). I've used temperature 1.0, top_p = 0.9, repetition_penalty = 1.03, max_output_length=256 for the experiments.

**Comment by shaoyijia** [comment]
Thanks for your quick response! Then the behavior I encountered makes sense.

Yeah, I think supporting document-level evaluation will be super cool. I feel that chat models have been actively studied but writing long grounded documents (e.g., expository writing) are relatively less studied.

I know there are works like [LLaMA-32k](https://huggingface.co/togethercomputer/LLaMA-2-7B-32K). Besides the base model context length, another constraint of extending Prometheus to support longer text may be your training dataset. It seems like your dataset construction pipeline can be directly used to produce longer samples but I am unsure how it will work in practice.

Yeah, I'm happy to keep in touch. I feel if in some certain tasks, people start to use your models as a "default metric", then this work will become super impactful!!
